extensions/wikia/CreateNewWiki/tasks/ConfigureWikiFactory.php

```diff
@@ -0,0 +1,262 @@
+		}
+
+		if ( CreateDatabase::ACTIVE_CLUSTER ) {
+			wfGetLBFactory()->sectionsByDB[$dbName] = $wikiFactoryVariables['wgDBcluster'] = CreateDatabase::ACTIVE_CLUSTER;
+		}
+
+		return $wikiFactoryVariables;
+	}
+
+	public function getVariablesFromDB( $sharedDBW, $wikiFactoryVariables ) {
+		wfProfileIn( __METHOD__ );
+
+		$oRes = $sharedDBW->select(
+			"city_variables_pool",
+			[ "cv_id, cv_name" ],
+			[ "cv_name in ('" . implode( "', '", array_keys( $wikiFactoryVariables ) ) . "')" ],
+			__METHOD__
+		);
+
+		$wikiFactoryVarsFromDB = [ ];
+
+		while ( $oRow = $sharedDBW->fetchObject( $oRes ) ) {
+			$wikiFactoryVarsFromDB[$oRow->cv_name] = $oRow->cv_id;
+		}
+		$sharedDBW->freeResult( $oRes );
+
+		wfProfileOut( __METHOD__ );
+
+		return $wikiFactoryVarsFromDB;
+	}
+
+	private function setVariables( $sharedDBW, $cityId, $wikiFactoryVariablesFromDB, $wikiFactoryVariables ) {
+		wfProfileOut( __METHOD__ );
+		foreach ( $wikiFactoryVariables as $variable => $value ) {
+			/**
+			 * first, get id of variable
+			 */
+			$cv_id = 0;
+			if ( isset($wikiFactoryVariablesFromDB[$variable]) ) {
+				$cv_id = $wikiFactoryVariablesFromDB[$variable];
+			}
+
+			/**
+			 * then, insert value for wikia
+			 */
+			if ( !empty($cv_id) ) {
+				$sharedDBW->insert(
+					"city_variables",
+					array(
+						"cv_value" => serialize( $value ),
+						"cv_city_id" => $cityId,
+						"cv_variable_id" => $cv_id
+					),
+					__METHOD__
+				);
+			}
+		}
+
+		$sharedDBW->commit( __METHOD__ ); // commit shared DB changes
+		wfProfileOut( __METHOD__ );
+	}
+
+	/**
+	 * "calculates" the value for wgUploadDirectory
+	 *
+	 * @access private
+	 * @author Piotr Molski (Moli)
+	 *
+	 * @param $name string base name of the directory
+	 * @param $language string language in which wiki will be created
+	 *
+	 * @return string
+	 */
+	public function prepareDirValue( $name, $language ) {
+		wfProfileIn( __METHOD__ );
+
+		$this->debug( implode( ":", [ "CreateWiki", __METHOD__ . "Checking {$name} folder" ] ) );
+
+		$isExist = false;
+		$suffix = "";
+		$dir_base = self::sanitizeS3BucketName( $name );
+		$prefix = strtolower( substr( $dir_base, 0, 1 ) );
+		$dir_lang = (isset($language) && $language !== "en")
+			? "/" . strtolower( $language )
+			: "";
+
+		while ( $isExist == false ) {
+			$dirName = self::IMGROOT . $prefix . "/" . $dir_base . $suffix . $dir_lang . "/images";
+
+			if ( self::wgUploadDirectoryExists( $dirName ) ) {
+				$suffix = rand( 1, 9999 );
+			} else {
+				$dir_base = $dir_base . $suffix;
+				$isExist = true;
+			}
+		}
+
+		$this->debug( implode( ":", [ "CreateWiki", __METHOD__, "Returning '{$dir_base}'" ] ) );
+		wfProfileOut( __METHOD__ );
+		return $dir_base;
+	}
+
+	public static function wgUploadDirectoryExists( $sDirectoryName ) {
+		wfProfileIn( __METHOD__ );
+		$iVarId = WikiFactory::getVarIdByName( 'wgUploadDirectory' );
+
+		// Crash immediately if $iVarId is not a positive integer!
+		\Wikia\Util\Assert::true( $iVarId );
```

Comment:
hmm will this add some logs? Maybe it would be better to log an error and throw an exception?